src/objects.js

```diff
@@ -130,6 +146,70 @@ export class Page {
   mapData(func) {
     return new Page(this.data.map(func), this.before, this.after)
   }
+}
+
+/** Used to iterate over the pages of a query. */
+export class PageIterator {
+  /**
+   * @param {Client} client
+   * @param {number} opts.pageSize Number of elements in each page.
+   * @param opts.mapLambda
+   *   Mapping query (made by {@link lambda}) applied to each element of each page.
+   * @param {function} opts.map
+   *   Mapping JavaScript function used on each page element.
+   */
+  constructor(client, set, opts={}) {
+    this.client = client
+    this.set = set
+    /** Set to `true` when there are no elements left to page through. */
+    this.done = false
+
+    Object.assign(this, applyDefaults(opts, {
+      pageSize: null,
+      mapLambda: null,
+      map: null
+    }))
 
-  // TODO: setIterator is async, so need a different way to do that...
+    // Change nulls for pageSize to `undefined` because
+    // `undefined` is automatically removed from queries by JSON.stringify.
+    if (this.pageSize === null)
+      this.pageSize = undefined
+
+    this.direction = undefined
+    this.cursor = undefined
+  }
+
+  /**
+   * Moves this iterator to the next page and returns its data.
+   * May set `this.done` to `true`.
+   */
+  async next() {
```

Comment:
I'm not sure about the JS world, but modeling something simple after Scala's http://www.scala-lang.org/api/current/index.html#scala.collection.immutable.Stream probably would be fine